docs/architecture/data-plane-osd-cluster-dynamic-scaling.md

```diff
@@ -6,18 +6,181 @@ This documents explains how Data Plane OSD Cluster Dynamic Scaling functionality
+
+```yaml
+  standard:
+    num_reserved_instances: 7
+    base_streaming_unit_size: x1
+```
+
+>NOTE: To disable pre-warming for a given instance type, set the `num_reserved_instances` to `0`. 
+>NOTE: To disable pre-warming completely, set the configuration as empty `{}` or omit it completely. 
+
+### Reserved ManagedKafka 
+
+For each data plane cluster, the Fleet manager returns a list of reserved managed Kafkas depending on the instance type that's supported by the cluster. 
+The number of generated reserved managed Kafkas in the cluster is the sum of the specified number of reserved instances among all instance types supported by the cluster. If the cluster is not in `ready` status the result is an empty list. Generated reserved Kafka names have the following naming schema:
+`reserved-Kafka-<instance_type>-<Kafka_number>` where Kafka_number goes from `1..<num_reserved_instances>_for_the_given_instance_type>`
+Each generated reserved Kafka has a namespace equal to its name and it contains the `bf2.org/deployment: reserved` label which identifies it as a reserved deployment. The Strimzi, Kafka and Kafk IBP versions are always set to the latest version.  
 
 ## OSD Cluster creation and deletion
 
-<!--- TODO -->
+### OSD cluster capacity information
+
+For each cluster, the fleet manager stores the dynamic capacity information per instance type. These information are useful for evaluation the Kafka placement strategy onto a cluster and for cluster auto creation / deletion evaluation logic.
+
+The struct is of the form:
+```yaml
+instance_type:
+  max_nodes: <integer> # The maximum number of nodes for this instance type. This config is read from the dynamic_scaling_config.go and initialised once the cluster is in provisioned stage after when the machine pool has been created.
+  max_units: <integer> # The maximum number of streaming units that can be provisioned on the given max_nodes. This is computed and sent by the Fleetshard sync
+  remaining_units: <integer> # The number of reaminign streaming units that can be provisioned. This is sent by the Fleetshard sync
+```
+
+For example:
+
+For a cluster that supports both standard and developer instances, we could have the following information stored in the database. 
+```yaml
+{
+ standard: {
+   max_nodes: 30
+   max_units: 10
+   remaining_units: 4
+ },
+ developer: {
+   max_nodes: 1,
+   max_units: 10,
+   remaining_units: 8
+ }
+}
+```
+
+1. To retrieve the information for a give cluster:
+```go
+cluster.RetrieveDynamicCapacityInfo()
+```
+
+the above method has to be called every time you want to check the information.
+
+2. To set the infomation
+```go
+cluster.SetDynamicCapacityInfo(map[string]DynamicCapacityInfo{})
+```
+
+### OSD Cluster creation
```

Comment:
thanks @miguelsorianod as much of this writing is from the code comments you left. Superb work!